### PR TITLE
Rtt 13 activity log module initial

### PR DESCRIPTION
**Pull request by eduranperez**

### CODE REVIEW

File: src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    @RequestMapping(value = "/api/workgroups/{workgroupId}/modules/{moduleName}/auditLogs", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
+    public List<AuditLog> getAuditLogs(@PathVariable long workgroupId,
+                                       @PathVariable String moduleName,
+                                       HttpServletResponse httpResponse) {
+
+        Workgroup workgroup = workgroupService.findOneById(workgroupId);
+
+        if (workgroup == null) {
```
Comment (flcli) [inline src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java:35]: What if `workgroupService.findOneById()` can't find anything?

File: src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @RequestMapping(value = "/api/workgroups/{workgroupId}/years/{year}/modules/{moduleName}/auditLogs", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
+    public List<AuditLog> getAuditLogsByYear(@PathVariable long workgroupId,
+                                       @PathVariable long year,
+                                       @PathVariable String moduleName,
+                                       HttpServletResponse httpResponse) {
+
+        Workgroup workgroup = workgroupService.findOneById(workgroupId);
+
+        if (workgroup == null) {
```
Comment (flcli) [inline src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java:55]: See above

File: src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java (hunk 1 above)
Comment (jaroldwong) [reply src/main/java/edu/ucdavis/dss/ipa/api/entities/AuditLogController.java:35]: 404 not found. Front end shows log modal without entries.

File: src/main/java/edu/ucdavis/dss/ipa/repositories/AuditLogRepository.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import edu.ucdavis.dss.ipa.entities.AuditLog;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface AuditLogRepository extends CrudRepository<AuditLog, Long>  {
+    @Query( " SELECT DISTINCT al" +
```
Comment (jaroldwong) [inline src/main/java/edu/ucdavis/dss/ipa/repositories/AuditLogRepository.java:11]: does this need to be a custom query? or would JPA suffice?

**Pull request merged**